fallout/mob/falloutmobs.dm

```diff
@@ -0,0 +1,665 @@
+	icon_dead = "[icon_state]_dead"
+	var/matrix/bambinoscale = matrix()
+	if(is_young == TRUE)
+		bambinoscale.Scale(0.7, 0.7)
+		transform = bambinoscale
+		return
+	if(smallasslad == TRUE)
+		//makes them small ass lads look all funky n shi
+		pixel_x = rand(-6, 6)
+		pixel_y = rand(0, 10)
+	if(tame)
+		if(milkable == TRUE)
+			chems = new()
+			return . = ..()
+		if(eggable == TRUE)
+			eggsleft = 0
+	else
+		return . = ..()
+
+/mob/living/simple_animal/hostile/retaliate/fallout/Destroy()
+	if(milkable == TRUE)
+		qdel(chems)
+		chems = null
+		return ..()
+	return . = ..()
+
+//hunger and baby grow/birth/speed shitcode
+/mob/living/simple_animal/hostile/retaliate/fallout/Life()
+	. = ..()
+	if(enemies.len && prob(10))
+		enemies = list()
+		LoseTarget()
+		src.visible_message("<span class='notice'>[src] calms down.</span>")
+		return
+	if(tame)
+		var/hungerrate = rand(0.2,0.3)
+		hunger -= hungerrate
+		return
+		if(hunger == 0)
+			hungerrate = 0
+			return
+	if(stat == CONSCIOUS && breedable == TRUE)
+		if((prob(3) && has_young))
+			has_young++
+		if(has_young > 10)
+			has_young = 0
+			visible_message("<span class='alertalien'>[src] has produced a baby.</span>")
+			new young_type(src.loc)
+		if(is_young)
+			if(prob(3))
+				is_young = 0
+				new adult_type(src.loc)
+				qdel(src)
+				visible_message("<span class='alertalien'>[src] has fully grown.</span>")
+				if(milkable == TRUE)
+					chems = new()
+				return
+	if(milkable == TRUE)
+		if(hunger == 0)
+			return
+		else
+			chems.generateChem()
+			return
+	if(eggable == TRUE)
+		if(!stat)
+			if(hunger >= 150 && eggsleft < 8)
+				eggsleft += rand(1, 4)
+				hunger -= 100
+			if(hunger == 0)
+				eggsleft = 0 //starvin now foo
+			else
+				return
+			return
+		if((!stat && prob(3) && eggsleft > 0) && egg_type)
+			visible_message("<span class='alertalien'>[src] [pick(EGG_LAYING_MESSAGES)]</span>")
+			eggsleft--
+			var/obj/item/E = new egg_type(get_turf(src))
+			E.pixel_x = rand(-6,6)
+			E.pixel_y = rand(-6,6)
+			return
+	if(tame == 1)
+		visible_message("<span class='alertalien'>[src] is now tamed.</span>")
+	if(saddled == TRUE)
+		saddled()
+		return
+	return
+
+/mob/living/simple_animal/hostile/retaliate/fallout/proc/saddled()
+	var/speedmodifier = speed/(hunger/maxhunger)
+	if(saddled == TRUE)
+		can_buckle = TRUE
+		buckle_lying = FALSE
+		add_overlay("[icon_state]_saddled")
+		var/datum/component/riding/D = LoadComponent(/datum/component/riding)
+		D.set_riding_offsets(RIDING_OFFSET_ALL, list(TEXT_NORTH = list(0, 8 + offsety), TEXT_SOUTH = list(0, 8 + offsety), TEXT_EAST = list(-2 - offsetx, 8 + offsety), TEXT_WEST = list(2 + offsetx, 8 + offsety)))
+		D.set_vehicle_dir_layer(SOUTH, RIDING_LAYER)
+		D.set_vehicle_dir_layer(NORTH, OBJ_LAYER)
+		D.set_vehicle_dir_layer(EAST, OBJ_LAYER)
+		D.set_vehicle_dir_layer(WEST, OBJ_LAYER)
+		D.keytype = /obj/item/key/lasso
+		D.drive_verb = "ride"
+		update_icon()
+		D.vehicle_move_delay = speedmodifier
```

Comment:
If you're gonna have 3 procs with the same code, it'd be best to make 1 proc that holds all the behavior. That way if you want to say change the keytype, you don't need to do it 3 times